main.go

```diff
@@ -17,103 +17,209 @@ limitations under the License.
-	flag.Parse()
-
-	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
-
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
-		HealthProbeBindAddress: probeAddr,
-		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "5a3ca514.codeflare.dev",
+	zapOptions.BindFlags(flag.CommandLine)
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&zapOptions)))
+
+	ctx := ctrl.SetupSignalHandler()
+
+	cfg := &config.CodeFlareOperatorConfiguration{
+		ClientConnection: &config.ClientConnection{
+			QPS:   pointer.Float32(50),
+			Burst: pointer.Int32(100),
+		},
+		ControllerManager: config.ControllerManager{
+			Metrics: config.MetricsConfiguration{
+				BindAddress: ":8080",
+			},
+			Health: config.HealthConfiguration{
+				BindAddress:           ":8081",
+				ReadinessEndpointName: "readyz",
+				LivenessEndpointName:  "healthz",
+			},
+			LeaderElection: &configv1alpha1.LeaderElectionConfiguration{},
+		},
+		MCAD: &mcadconfig.MCADConfiguration{},
+		InstaScale: &config.InstaScaleConfiguration{
+			Enabled: pointer.Bool(false),
+			InstaScaleConfiguration: instascaleconfig.InstaScaleConfiguration{
+				MaxScaleoutAllowed: 5,
+			},
+		},
+	}
+
+	kubeConfig, err := ctrl.GetConfig()
+	exitOnError(err, "unable to get client config")
+	if kubeConfig.UserAgent == "" {
+		kubeConfig.UserAgent = "codeflare-operator"
+	}
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	exitOnError(err, "unable to create Kubernetes client")
+
+	exitOnError(loadIntoOrCreate(ctx, kubeClient, namespaceOrDie(), configMapName, cfg), "unable to initialise configuration")
+
+	kubeConfig.Burst = int(pointer.Int32Deref(cfg.ClientConnection.Burst, int32(rest.DefaultBurst)))
+	kubeConfig.QPS = pointer.Float32Deref(cfg.ClientConnection.QPS, rest.DefaultQPS)
+	setupLog.V(2).Info("REST client", "qps", kubeConfig.QPS, "burst", kubeConfig.Burst)
+
+	mgr, err := ctrl.NewManager(kubeConfig, ctrl.Options{
+		Scheme:                     scheme,
+		MetricsBindAddress:         cfg.Metrics.BindAddress,
+		HealthProbeBindAddress:     cfg.Health.BindAddress,
+		LeaderElection:             pointer.BoolDeref(cfg.LeaderElection.LeaderElect, false),
+		LeaderElectionID:           cfg.LeaderElection.ResourceName,
+		LeaderElectionNamespace:    cfg.LeaderElection.ResourceNamespace,
+		LeaderElectionResourceLock: cfg.LeaderElection.ResourceLock,
+		LeaseDuration:              &cfg.LeaderElection.LeaseDuration.Duration,
+		RetryPeriod:                &cfg.LeaderElection.RetryPeriod.Duration,
+		RenewDeadline:              &cfg.LeaderElection.RenewDeadline.Duration,
 	})
-	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+	exitOnError(err, "unable to start manager")
+
+	mcadQueueController := mcad.NewJobController(mgr.GetConfig(), cfg.MCAD, &mcadconfig.MCADConfigurationExtended{})
+	if mcadQueueController == nil {
+		// FIXME
```

Comment:
Can you elaborate on this?

Reply:
I've updated the FIXME with more context.